### PR TITLE
185294347 enrollment filter options

### PR DESCRIPTION
**Pull request by ZachCMP**
https://www.pivotaltracker.com/story/show/185294347

Relies on frontend PR: https://github.com/greenriver/hmis-frontend/pull/291

### CODE REVIEW

File: db/warehouse_structure.sql [hunk 1]
```diff
@@ -122,116 +122,116 @@ CREATE TYPE public.record_type AS ENUM (
 
 CREATE FUNCTION public.service_history_service_insert_trigger() RETURNS trigger
     LANGUAGE plpgsql
-    AS $$
```
Comment (gigxz) [inline db/warehouse_structure.sql:125]: There are no db changes, so no need to commit this file.

File: drivers/hmis/app/graphql/types/hmis_schema/project.rb [hunk 2]
```diff
@@ -47,7 +47,7 @@ def self.configuration
     hud_field :date_deleted
     field :user, HmisSchema::User, null: true
     field :active, Boolean, null: false
-    enrollments_field without_args: [:project_types]
+    enrollments_field filter_omit: [:project_types], filter_type_name: 'EnrollmentsForProject'
```
Comment (gigxz) [inline drivers/hmis/app/graphql/types/hmis_schema/project.rb]: I guess I missed this last time, I thought this was going to look like the below now that a `filter_args` argument is accepted
```
enrollments_field filter_args: { omit: [:project_types], type_name: 'EnrollmentsForProject' }
```

File: drivers/hmis/app/models/hmis/filter/assessment_filter.rb [hunk 3]
```diff
@@ -14,8 +14,6 @@ def filter_scope(scope)
   protected
 
   def with_roles(scope)
-    return scope unless input.roles&.present?
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/filter/assessment_filter.rb:17]: Why is this removed?

File: drivers/hmis/app/models/hmis/hud/enrollment.rb [hunk 4]
```diff
@@ -57,6 +57,11 @@ class Hmis::Hud::Enrollment < Hmis::Hud::Base
 
   SORT_OPTIONS = [:most_recent, :household_id].freeze
 
+  SORT_OPTION_DESCRIPTIONS = {
+    most_recent: 'Most Recent First',
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/hud/enrollment.rb]: The "first" seems redundant, I think it can just be "Most Recent"

File: drivers/hmis/app/graphql/types/hmis_schema/has_enrollments.rb [hunk 5]
```diff
@@ -36,8 +43,9 @@ def resolve_enrollments(scope = object.enrollments, **args)
 
       private
 
-      def scoped_enrollments(scope, sort_order: :most_recent, enrollment_limit: nil, open_on_date: nil, project_types: nil, search_term: nil)
+      def scoped_enrollments(scope, sort_order: :most_recent, enrollment_limit: nil, open_on_date: nil, project_types: nil, search_term: nil, filters: nil)
```
Comment (gigxz) [inline drivers/hmis/app/graphql/types/hmis_schema/has_enrollments.rb]: Unused arguments can be removed

File: drivers/hmis/app/graphql/types/hmis_schema/has_enrollments.rb [hunk 6]
```diff
@@ -36,8 +43,9 @@ def resolve_enrollments(scope = object.enrollments, **args)
...
         scope = scope.viewable_by(current_user)
+        scope = scope.apply_filters(filters) if filters.present?
         scope = scope.not_in_progress if enrollment_limit == 'NON_WIP_ONLY'
         scope = scope.in_progress if enrollment_limit == 'WIP_ONLY'
         scope = scope.open_on_date(open_on_date) if open_on_date.present?
```
Comment (gigxz) [inline drivers/hmis/app/graphql/types/hmis_schema/has_enrollments.rb]: These and the 2 lines below can be removed

**Review by gigxz — COMMENTED** [review]
Looks good, left a few minor comments. Only thing blocking should be to delete the unused code.

**Comment by ZachCMP** [comment]
@gigxz All feedback should be addressed now

**Comment by gigxz** [comment]
node modules are being committed 👀   git ignore probably needs to be updated since https://github.com/greenriver/hmis-warehouse/pull/3102

can you remove them from the branch?

**Comment by ZachCMP** [comment]
@gigxz Okay, I removed it

**Review by gigxz — APPROVED** [review]

**Comment by gigxz** [comment]
I just made one change to include WIP Enrollments in the project type filter. Ready to merge

**Comment by eanders** [comment]
@gigxz this still has the DB structure file changes (which is probably fine), do you want to remove that or just go with it?

**Comment by gigxz** [comment]
Looks like it didn't break tests so I guess its fine
